### ISSUE TITLE
Image snapshot testing with Puppeteer

### ISSUE THREAD

**Issue by rolyp** [tgve/eatlas-template#48]
Snapshot (approval) testing is useful for easy refactoring, and well supported by Jest. It's not quite trivial to set up for us because the map view renders asynchronously, and so waiting on "Nothing to show" doesn't seem to be enough for deterministic behaviour. The following diff illustrates different states that both satisfy `expect(..).toContain("Nothing to show")`:

![image-name-diff](https://user-images.githubusercontent.com/121074/158341288-4eace248-1297-478e-83be-937bd51d4f2e.png)

Maybe we can find a UI element to wait on that only appears when the map is fully loaded.

- [x] install `jest-image-snapshot`
- [x] import `toMatchImageSnapshot` from `jest-image-snapshot` and extend Jest’s `expect` API
- <s>condition that over-approximates map being fully loaded</s>
  - <s>`waitUntil: "networkidle0"`</s> -- probably not very reliable
- <s>revisit snapshot tests in `tgvejs` (maybe no longer needed)</s>
- [x] set visibility of map and loader component to `hidden` (to avoid problem with async map content)
- [x] change Puppeteer viewport size to include more of the side desk
- [x] export `__snapshort` folder as build artifact
- [x] set error tolerance in screenshot diffs to 0.5%

See:

- [Snapshot testing with Puppeteer](https://loadfocus.com/blog/2018/02/visual-ui-regression-testing-with-jest-puppeteer-and-headless-chrome/)
- [Puppeteer options](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pagegotourl-options) for detecting navigation complete
- [Puppeteer wait until page is completely loaded](https://stackoverflow.com/questions/52497252/puppeteer-wait-until-page-is-completely-loaded)
- [Puppeteer to listen for map.on('load') from within Node](https://stackoverflow.com/questions/58104413/puppeteer-to-listen-for-map-onload-from-within-node)

**Comment by layik** [comment]
Shall we try `waitUntil: 'networkidle2'`?

**Comment by layik** [comment]
![test1](https://user-images.githubusercontent.com/408568/158387118-3be852c2-8137-46ee-8ade-1aa06e073df6.png)
![test2](https://user-images.githubusercontent.com/408568/158387121-93d7c404-259b-448e-91ac-749b559ffa2e.png)
![test3](https://user-images.githubusercontent.com/408568/158387124-06580848-211f-44fa-a737-9e9178b0da7f.png)
These are captured with `waitUntil: 'networkidle0'`, tried `2` almost identical.

**Comment by layik** [comment]
OK having played with many options on puppeteer, a solution which also contributes to overall tgvejs functionality is to generate the image snapshots without "basemap"s. This can be achieved by exporting another API called `mapStyle` and setting it to "No map". The results would be like. Please let me know @rolyp if you like me to send the commit (small one) to `tgve/tgvejs`.
![test1](https://user-images.githubusercontent.com/408568/158401871-c40838ba-0931-4d35-be3d-bfe9a37379c4.png)
![test3](https://user-images.githubusercontent.com/408568/158401910-748d55f2-6360-4427-b818-f540ed049649.png)

Leave it here, got enough about puppeteer and how we are trying to use it. Great work btw.
